### PR TITLE
chore: Test `lib/archive.ts`

### PR DESCRIPTION
**Pull request by joe-yeager**
## Description and Context

- Add tests for `lib/archive.ts`
- Fix a couple bugs discovered in the testing process with async methods not being awaited
- Fix a typo in error messages

## Who to Notify

@camden11 @brandenrodgers @kemmerle

### CODE REVIEW

File: lib/archive.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ async function copySourceToDest(
       if (!rootDir) {
         logger.debug(i18n(`${i18nKey}.copySourceToDest.sourceEmpty`));
         // Create the dest path if it doesn't already exist
-        fs.ensureDir(dest);
+        await fs.ensureDir(dest);
```
Comment (joe-yeager) [inline lib/archive.ts:98]: This is an async method unless a callback is provided.  [Docs](https://github.com/jprichardson/node-fs-extra/blob/master/docs/ensureDir.md#ensuredirdiroptionscallback)

File: lib/archive.ts [hunk 2]
```diff
@@ -121,10 +121,10 @@ async function copySourceToDest(
   return false;
 }
 
-function cleanupTempDir(tmpDir: string): void {
+async function cleanupTempDir(tmpDir: string): Promise<void> {
   if (!tmpDir) return;
   try {
-    fs.remove(tmpDir);
+    await fs.remove(tmpDir);
```
Comment (joe-yeager) [inline lib/archive.ts:127]: Same with this method.  [Docs](https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md)

File: lang/en.json [hunk 3]
```diff
@@ -10,15 +10,15 @@
         "init": "Extracting project source...",
         "success": "Completed project source extraction.",
         "errors": {
-          "write": "An error occured writing temp project source.",
-          "extract": "An error occured extracting project source."
+          "write": "An error occurred writing temp project source.",
+          "extract": "An error occurred extracting project source."
```
Comment (camden11) [inline lang/en.json:14]: I don't even think this is the first time we've spelled occurred incorrectly lol

**Review by camden11 — APPROVED** [review]
LGTM

**Pull request merged**